controllers/storageconsumer/storageconsumer_controller.go

```diff
@@ -145,6 +134,37 @@ func (r *StorageConsumerReconciler) reconcilePhases() (reconcile.Result, error)
 
 	if r.storageConsumer.GetDeletionTimestamp().IsZero() {
 
+		storageCluster, err := util.GetStorageClusterInNamespace(r.ctx, r.Client, r.namespace)
+		if err != nil {
+			return reconcile.Result{}, err
+		}
+
+		consumerConfigMap := &corev1.ConfigMap{}
+		consumerConfigMap.Name = r.storageConsumer.Name
+		if r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name != "" {
+			consumerConfigMap.Name = r.storageConsumer.Spec.ResourceNameMappingConfigMap.Name
+		}
+		consumerConfigMap.Namespace = r.namespace
+		if err = r.get(consumerConfigMap); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		isControllerOwner := false
+
+		if owner := util.FindControllerOwnerRef(consumerConfigMap); owner != nil && owner.UID == r.storageConsumer.UID {
+			isControllerOwner = true
+		}
+
+		consumerResources := util.WrapStorageConsumerResourceConfigMap(consumerConfigMap)
+
+		if err := r.reconcileCephRadosNamespace(isControllerOwner, storageCluster, consumerResources); err != nil {
+			return reconcile.Result{}, err
+		}
+
+		if err := r.reconcileCephFilesystemSubVolumeGroup(isControllerOwner, storageCluster, consumerResources); err != nil {
+			return reconcile.Result{}, err
+		}
```

Comment:
Why send the `isControllerOwner` to the functions instead of just having an branch on the outside?